### PR TITLE
test: use transitionend instead of settimeout

### LINKED ISSUE #447: Test Failure: style applies tranform as transition on remove FAILED
I noticed below test failure upon running `npm test` on my machine. A PR is coming up soon.
```
> karma start

05 09 2019 18:19:31.761:INFO [compiler.karma-typescript]: Compiling project using Typescript 3.3.4000
05 09 2019 18:19:36.427:INFO [compiler.karma-typescript]: Compiled 26 files in 4606 ms.
05 09 2019 18:19:37.335:INFO [bundler.karma-typescript]: Bundled imports for 26 file(s) in 404 ms.
05 09 2019 18:19:38.801:INFO [karma-server]: Karma v3.1.4 server started at http://0.0.0.0:9876/
05 09 2019 18:19:38.803:INFO [launcher]: Launching browsers Chrome, Firefox with concurrency unlimited
05 09 2019 18:19:38.811:INFO [launcher]: Starting browser Chrome
05 09 2019 18:19:38.829:INFO [launcher]: Starting browser Firefox
05 09 2019 18:19:40.721:INFO [Chrome 76.0.3809 (Windows 10.0.0)]: Connected on socket a2jgjAhxrFJuEPpKAAAA with id 27571023
................................................................................
...................................................................
Chrome 76.0.3809 (Windows 10.0.0): Executed 147 of 147 SUCCESS (1.075 secs / 0.862 secs)
05 09 2019 18:19:44.824:INFO [Firefox 69.0.0 (Windows 10.0.0)]: Connected on socket bzlBMsLmXtLqFFD7AAAB with id 22765761
................................................................................
.....................................................
Firefox 69.0.0 (Windows 10.0.0) style applies tranform as transition on remove FAILED
        uncaught exception: AssertionError: 1 === 0 (:0)
.............
Firefox 69.0.0 (Windows 10.0.0): Executed 147 of 147 (1 FAILED) (0.911 secs / NaN secs)
TOTAL: 1 FAILED, 293 SUCCESS
npm ERR! Test failed.  See above for more details.
```

### PR DESCRIPTION
**Pull request by kl2134**
Fixes #447

```
> karma start

05 09 2019 18:35:25.811:INFO [compiler.karma-typescript]: Compiling project using Typescript 3.3.4000
05 09 2019 18:35:31.057:INFO [compiler.karma-typescript]: Compiled 26 files in 5146 ms.
05 09 2019 18:35:32.028:INFO [bundler.karma-typescript]: Bundled imports for 26 file(s) in 469 ms.
05 09 2019 18:35:33.634:INFO [karma-server]: Karma v3.1.4 server started at http://0.0.0.0:9876/
05 09 2019 18:35:33.637:INFO [launcher]: Launching browsers Chrome, Firefox with concurrency unlimited
05 09 2019 18:35:33.655:INFO [launcher]: Starting browser Chrome
05 09 2019 18:35:33.675:INFO [launcher]: Starting browser Firefox
05 09 2019 18:35:36.368:INFO [Chrome 76.0.3809 (Windows 10.0.0)]: Connected on socket Y5ZOZ2SYM9T37QqGAAAA with id 8769878
................................................................................
...................................................................
Chrome 76.0.3809 (Windows 10.0.0): Executed 147 of 147 SUCCESS (0.452 secs / 0.249 secs)
05 09 2019 18:35:40.124:INFO [Firefox 69.0.0 (Windows 10.0.0)]: Connected on socket AcbDDuIsbUD3dl86AAAB with id 5279393
................................................................................
...................................................................
Firefox 69.0.0 (Windows 10.0.0): Executed 147 of 147 SUCCESS (0.586 secs / 0.248 secs)
TOTAL: 294 SUCCESS
```

Mentored by @mightyiam.

### CODE REVIEW

File: test/style.js [hunk 1]
```diff
@@ -118,19 +118,20 @@ describe('style', function() {
   });
   it('applies tranform as transition on remove', function(done) {
     var btn = h('button', { style: {
-        transition: 'transform 0.5s',
+        transition: 'transform 0.1s',
```
Comment (mightyiam) [inline test/style.js]: Shortening the animation was my suggestion. For the sake of a faster test suite.

File: test/style.js [hunk 2]
```diff
@@ -118,19 +118,20 @@ describe('style', function() {
...
-    setTimeout(function () {
-        assert.strictEqual(document.querySelectorAll('button').length, 0);
-        done();
-    }, 700);
+    const button = document.querySelector('button');
+    assert.notStrictEqual(button, null);
+    button.addEventListener('transitionend', function() {
+      assert.strictEqual(document.querySelector('button'), null);
+      done();
+    });
```
Comment (mightyiam) [inline test/style.js]: So, this seems to remove the race condition and also should shorten the test duration by about 600ms.

File: test/style.js (hunk 1 above)
Comment (paldepind) [reply test/style.js]: That's a good idea :+1:

**Comment by paldepind** [comment]
This looks great. Definitely a more appropriate way of doing it. Instead of just waiting and hoping that the transition is done we actually wait listen for a signal that it is done.

**Comment by paldepind** [comment]
It would be interesting to know if this is what caused some of the CI browser tests to fail. That is, we could try reinstating some of the tests commented out in #444 and see if some of them now succeed.

**Comment by mightyiam** [comment]
Thank you, @paldepind. Tried in #449. Seems like there's more to those failures than this.

**Pull request merged**